Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -737,9 +737,9 @@ void LoadPSIMuonBin::readInTemperatureFileHeader(const std::string &contents) {
   std::string line = "";
   for (const auto charecter : contents) {
     if (charecter == '\n') {
-      if (line[0] == '!' && lineNo > uselessLines) {
```

Comment:
Would this read more easily if we did the `empty` check first & once then continued?